alf/config_helpers.py

```diff
@@ -223,7 +228,8 @@ def parse_config(conf_file, conf_params):
         _is_parsing = False
 
     # Create the global environment and initialize random seed
-    get_env()
+    if create_env:
```

Comment:
This won't solve the problem for all configuration. Some will call alf.get_observation_spec() or alf.get_action_spec() inside the configuration. Both function will call get_env()

Reply:
I see what you mean. The tricky part is that at this moment when environment is being created, even the parent process cannot do `get_observation_spec` yet because of that `env = get_env()` call. Let me think of a solution. 

The "simple fix" might be not fixing it, i.e. just throw exception for the cases when `get_env()` gets called during config parse in the subprocesses ... which at least stops the infinite recursion.

Reply:
A viable approach might be hacking the `create_environment` function with pre configs. Let me do some experiment.

Reply:
I thought about another solution, which is to create the environment at main process, get the action/observation spec and send them to the subprocess.

However, this defeats the purpose that we do not want to construct the environment at the main process (e.g. mujoco at this moment will create the camera and mess up EGL context ...)